tests/Doctrine/Tests/ORM/Functional/Ticket/GH10049Test.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @requires PHP 8.1
+ */
+class GH10049Test extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            GH10049Child::class,
+            GH10049Parent::class
+        );
+    }
+
+    public function testInheritedReadOnlyPropertyValueIsSet(): void
+    {
+        $child = new GH10049Child(10049);
+        $this->_em->persist($child);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $entity = $this->_em->find(GH10049Child::class, 10049);
+
+        self::assertInstanceOf(GH10049Child::class, $entity);
+        self::assertSame(10049, $entity->id);
+    }
+}
+
+/**
+ * @ORM\MappedSuperclass
+ */
+class GH10049Parent
+{
+    public function __construct(
+        /**
+         * @ORM\Id
+         * @ORM\Column(type="integer")
+         */
+        public readonly int $id
```

Comment:
If you're going to use PHP8-specific syntax, you should move this class to its own file. An alternative would be to move the test file in a directory configured with https://docs.phpunit.de/en/10.1/configuration.html#the-testsuite-element

Reply:
Okay, opened #10683